### DISCUSSION TITLE
How to format sql + args into a complete query?

### DISCUSSION THREAD

**Discussion by tymbaca** [jackc/pgx#2049] [Q&A]
Is there a way to just format `sql string, args ...interface{}` into final query (as string or []byte)? 

I'm playing around with queries with `time.Time` and `time.Duration` arguments and want to inspect the final result

**Comment by jackc** [comment]
Unless you are using the simple protocol, the arguments are not interpolated into the query text. The are passed to PostgreSQL as separate parameters.

If you are using the simple protocol then there's no publicly available interface for that logic. The package that performs the sanitization but is private (`github.com/jackc/pgx/v5/internal/sanitize`). You could use the tracing / logging infrastructure to see the final query though.
